R/mrgindata.R

```diff
@@ -81,45 +81,62 @@ convert_character_cmt <- function(data, mod) {
   return(data)
 }
 
-signal_drop <- function(dm, x, to_signal, context) {
-  drop <- setdiff(names(x), dimnames(dm)[[2]])
-  drop <- intersect(drop, to_signal)
-  for(d in drop) {
+check_dropped_cols <- function(dm, x, check, context) {
+  new_col_set <- dimnames(dm)[[2]]
+  old_col_set <- names(x)
+  drop <- old_col_set[!old_col_set %in% new_col_set]
+  drop <- drop[drop %in% check]
+  if(!length(drop)) return(invisible(NULL))
+  body <- vector(mode = "character", length = length(drop))
+  names(body) <- rep("x", length(body))
+  for(i in seq_along(drop)) {
+    d <- drop[i]
     type <- paste0(class(x[[d]]), collapse = ",")
-    msg <- c(context, " dropped column: ", d, " (", type, ")")
-    message(msg)  
+    body[i] <- paste0(context, " column: ", d, " (", type, ")")
   }
-  invisible(NULL)
+  abort(
+    message = "Found input data that cannot be used for simulation", 
+    body = body, 
+    call = caller_env()
+  )
 }
 
 ##' Validate and prepare a data sets for simulation
 ##'
-##' This function is called by mrgsim.  Users may also call this function
-##' to pre-validate data when the same data set is used for repeated 
-##' simulation.
+##' This function is called by [mrgsim()] and friends.  Users may also call 
+##' this function to pre-validate data when the same data set is used for 
+##' repeated simulation.
 ##'
-##' @param x data.frame or matrix
-##' @param m a model object
-##' @param verbose logical
-##' @param quiet if \code{TRUE}, messages will be suppressed
+##' @param x data.frame or matrix.
+##' @param m a model object.
+##' @param verbose logical.
+##' @param quiet if `TRUE`, messages will be suppressed.
+##' 
+##' @details
```

Comment:
This is mostly called under the hood; but I filled out details since we're referencing the function name in the error message and probably good to get more of this written down somewhere.